Privacy policy

cwillisf opened the pull request:
### Resolves

Resolves #97

### Proposed Changes

Primary changes:

- Add a new window to display a built-in privacy policy without needing Internet access
- Move the existing "About" window activation from the "Scratch" logo to a new button in the menu bar
- Open the new privacy policy from either the "About" window or from the privacy policy link on the telemetry opt-in/out window

Supporting changes:

- Add necessary configuration and dependencies to support including CSS files from JSX files like we do in `scratch-gui`
- Previously the "About" window had a special handler to open links in the system browser. That has now moved to be shared across windows.
- Minor cleanup, especially in `app.jsx`

### Reason for Changes

There were several problems with our privacy policy in `scratch-desktop`:
- The Microsoft Store rejected the most recently submitted version of this application based on their privacy policy requirements, specifically in section 10.5.1 here: <https://docs.microsoft.com/en-us/windows/uwp/publish/store-policies#105-personal-information>
  - Note that `scratch-desktop` is a "desktop bridge" application as referenced in that policy.
- Previously we had a link to <https://scratch.mit.edu/privacy_policy> on the telemetry dialog, but nowhere else. That is, once the telemetry dialog is dismissed there's no way to view the privacy policy.
- That privacy policy is designed for our web site and doesn't fit this application very well. That's true of both the wording and the fact that it's on our web site, since `scratch-desktop` is designed to be used without an Internet connection some or all of the time.

These changes resolve all of those problems and will hopefully allow us to release a new version of `scratch-desktop` on the Microsoft Store.

### Legal note

We're working with some lawyers to review the privacy policy, so it may change based on their feedback. If you'd rather wait for final approval of the privacy policy before reviewing the code I'm OK with that, though I don't expect any policy changes to affect the rest of the code.

Commit:
move About window styling into a CSS file

Commit:
add new window for privacy policy

Commit:
open external links in system browser

Commit:
app.jsx cleanup

Commit:
add privacy policy link to 'about' dialog

Commit:
view built-in privacy policy from telemetry dialog

cwillisf commented on src/renderer/privacy.jsx:
```diff
@@ -0,0 +1,62 @@
+                href="https://scratch.mit.edu/contact-us/"
+                target="_blank"
+                rel="noopener noreferrer"
+            >contact us</a>.
```
I think it's OK for this to be a web site link since you won't be able to contact us unless you have Internet access anyway.

rschamp reviewed (approved):
These changes look good to me as far as I understand them. I am assuming we will do a playtest to check functionality across OS's.

Commit:
add script to merge scratch-gui deps

This is in preparation for having webpack build scratch-gui source
instead of consuming its build output. This script merges the
dependencies listed in package-deps.json with those listed in
scratch-gui's package.json, using the package-deps.json version in case
of conflict, then writes the result into the `devDependencies` section
of scratch-desktop's package.json.

Commit:
npm run rebuild-deps

Commit:
build scratch-gui from source instead of using its build output

Commit:
assume direct control of webpack config

Instead of overriding rules we don't want by matching the test exactly,
we now explicitly delete rules we don't want before adding any of our
custom rules.

Commit:
update library paths

Commit:
write configs to files in dist/ for debugging purposes

Commit:
print a warning if rebuild-deps changes deps

Commit:
move ReactDOM.render() into index.js

This also means we no longer need to disable eslint's
"no-unused-expressions" rule for each route in index.js

Commit:
reduce webpack output on CI

cwillisf commented:
Closing this for now to reduce email notifications... I'll reopen or make a new PR once it's ready for review again.
